### PR TITLE
Pin Zheng current event in all agenda highlight

### PR DESCRIPTION
**Pull request by Ducksss**
# Introduction
The all agenda page will now include a blinking "Now" Tag to highlight the event that is currently active beside the accordion title. This feature is to help visitors in finding events that are currently active in a programmatic manner. The feature will also refresh itself every 20 seconds to automatically remove and add the "Now" Tag from the headers/titles.

- Example of the "Now" Tag in headers / titles
Screenshot 2022-07-19 at 2 07 18 PM

- Example of how the app automatically re-renders every 20 seconds with animated blinking effects with the banner (note the "Now" Status is gone for the first event after the event is over while the second event now has the "Now" Status. 
https://user-images.githubusercontent.com/58126222/179676806-9189b7b8-ed0e-4328-9896-cff4cb2fa6da.mov

## File changes
- _data/stackx2022dsc.yml (Modified timeslot_metadata into a to and from date time)
- apps/src/pages/AllSessionsApp.vue (Vue Application that introduces the now tag and code for blinking animations)
- search/dataScienceConnect.json (Modified to accomodate for timeslot_metadata new strucutre)
- apps/src/sass/components/_animation.scss (Animation styling)
- apps/src/sass/components/_all.scss (Importing _animation.scss)

### CODE REVIEW

**Review by chewyixin99 — COMMENTED** [review]
looks good :-)

File: apps/src/pages/AllSessionsApp.vue [hunk 1]
```diff
@@ -117,7 +130,11 @@
                   </div>
                 </div>
                 <!-- Time and category-->
-                <small>{{ item.timeslot }} / {{ item.category.type }} </small>
```
[the remark this answers is not in the document]
Comment (Ducksss) [reply apps/src/pages/AllSessionsApp.vue:120]: The timeslot variable is used for event tracks, etc., outside of the all agenda page. The publishers' current templates (used in data science connect 2022 and stack x cloud 2021) do not support the use of timeslot metadata. Though, it is a great idea to revamp the other templates to support timeslot_metadata which is more comprehensive and have more information esp when parsing.

 such as https://www.developer.tech.gov.sg/communities/events/conferences/stack-x-data-science-connect-2022/agenda/day-1-programme and https://www.developer.tech.gov.sg/communities/events/conferences/stack-x-data-science-connect-2022/agenda/day-1-programme

File: search/dataScienceConnect.json [hunk 2]
```diff
@@ -32,33 +32,36 @@
...
-          {%- if forloop.last -%}{}{%- endif -%}
+          {
+              "title": "{{ document.title | remove: '"' }}",
+              "content": "{{ document.desc | remove: '>' | remove: '"' | strip }}",
+              "targetGroup": "{{ target_group_joined | default : "" }}",
+              "category": {
+                "type" : "{{ document.type | default : "" }}",
+                "icon" : "{{- "/assets/icons/" | append: icons[category_pointer] | append: ".svg"| default : "/assets/icons/star.svg" -}}"
+              },
+              "timeslot": "{{ document.timeslot | default : "" }}",
```
[the remark this answers is not in the document]
Comment (Ducksss) [reply search/dataScienceConnect.json:43]: Great point, will remove the timeslot variable from the search/dataScienceConnect.json file

**Comment by Ducksss** [comment]
@chewyixin99 I decided to recode a few parts of the logic while I was at it. The previous implementation would only evaluate the first event isActive attribute to display the flashing "Now" Icon. After thinking about how some events starts at the same time but ends at diff intervals, i.e. event1 starts at t0 -> t60 while event2 starts at t0 -> t70, I've adjusted the logic to account for it, i.e. t0 -> t70, if they share a common start point, i.e., t0.

Screenshot 2022-07-20 at 11 15 01 PM

Screenshot 2022-07-20 at 11 24 46 PM

**Comment by chewyixin99** [comment]
image

Saw that you changed the implementation slightly. Was wondering if having the "Now" icon within the card, next to the title would be better since it's possible for multiple events to start at the same time but end at different times like you said. Having the "Now" icon on the grey banner might be slightly misleading. Just my 2cents!

**Comment by Ducksss** [comment]
> In

okies, thats a good suggestion. I will ask the UI/UX team for their considerations.

**Review by chewyixin99 — APPROVED** [review]
looks good

**Pull request merged**